editor/plugins/org.fusesource.ide.project/src/org/fusesource/ide/project/camel/maven/CamelProjectConfigurator.java

```diff
@@ -205,4 +230,94 @@ private String inferCamelVersionFromDependencies(MavenProject mavenProject) {
 	private boolean isKnownFuseGroup(Artifact artifact) {
 		return (artifact.getGroupId().startsWith("org.jboss.quickstarts.fuse."));
 	}
+	
+	/**
+	 * checks whether the pom contains any apache camel dependency
+	 * 
+	 * @param m2Project
+	 * @return
+	 */
+	private boolean checkCamelDependencies(MavenProject m2Project) {
+		for (Dependency dep : m2Project.getDependencies()) {
+			if (dep.getGroupId().equalsIgnoreCase("org.apache.camel") && 
+				dep.getArtifactId().toLowerCase().startsWith("camel-")) {
+				return true;
+			}
+		}
+		return false;
+	}
+	
+	/**
+	 * checks whether the project contains any camel context xml file
+	 * 
+	 * @param project
+	 * @return
+	 */
+	private boolean checkCamelContextsExist(IProject project) throws CoreException {
+		return findFiles(project);
+	}
+	
+	/**
+	 * adds the fuse camel and java nature to the project if needed
+	 * 
+	 * @param project
+	 */
+	private void addFuseNature(IProject project, IProgressMonitor monitor) throws CoreException {
+		boolean camelNatureFound = false;
+		boolean javaNatureFound = false;
+		IProjectDescription projectDescription = project.getDescription();
+		String[] ids = projectDescription.getNatureIds();
+		for (String id : ids) {
```

Comment:
we might replace with Java 8 stream:
javaNatureFound = Arrays.stream(ids).anyMatch(s -> JavaCore.NATURE_ID.equals(s));
camelNatureFound = Arrays.stream(ids).anyMatch(s -> RiderProjectNature.NATURE_ID.equals(s));